prospero-cli/src/main/java/org/wildfly/prospero/cli/RepositoryDefinition.java

```diff
@@ -32,90 +31,67 @@
-                } else {
-                    repoId = "temp-repo-" + i;
-                    repoUri = parseRepositoryLocation(repoInfo);
+            if (repoInfo.contains("::")) {
+                final String[] parts = repoInfo.split("::");
+                if (parts.length != 2 || parts[0].isEmpty() || parts[1].isEmpty()) {
+                    throw CliMessages.MESSAGES.invalidRepositoryDefinition(repoInfo);
                 }
-                repositories.add(new Repository(repoId, repoUri));
-            } catch (URISyntaxException e) {
-                logger.error("Unable to parse repository uri + " + repoInfo, e);
-                throw CliMessages.MESSAGES.invalidRepositoryDefinition(repoInfo);
+                repoId = parts[0];
+                repoUri = parseRepositoryLocation(parts[1]);
+            } else {
+                if (StringUtils.isBlank(repoInfo)) {
+                    throw CliMessages.MESSAGES.invalidRepositoryDefinition(repoInfo);
+                }
+                repoId = "temp-repo-" + i;
+                repoUri = parseRepositoryLocation(repoInfo);
             }
 
+            repositories.add(new Repository(repoId, repoUri));
         }
         return repositories;
     }
 
-    private static String parseRepositoryLocation(String repoLocation) throws URISyntaxException, ArgumentParsingException {
-        if (!isRemoteUrl(repoLocation) && !repoLocation.isEmpty()) {
-            // the repoLocation contains either a file URI or a path
-            // we need to convert it to a valid file IR
-            repoLocation = getAbsoluteFileURI(repoLocation).toString();
-        }
-        if (!isValidUrl(repoLocation)){
-            throw CliMessages.MESSAGES.invalidRepositoryDefinition(repoLocation);
-        }
-        return repoLocation;
-    }
-
-    private static boolean isRemoteUrl(String repoInfo) {
-        return repoInfo.startsWith("http://") || repoInfo.startsWith("https://");
-    }
-
-    private static boolean isValidUrl(String text) {
+    private static String parseRepositoryLocation(String location) throws ArgumentParsingException {
+        URI uri;
         try {
-            new URL(text);
-            return true;
-        } catch (MalformedURLException e) {
-            return false;
+            uri = new URI(location);
+            if (("file".equals(uri.getScheme()) || StringUtils.isBlank(uri.getScheme()))
+                    && StringUtils.isBlank(uri.getHost())) {
+                // A "file:" URI with an empty host is assumed to be a local filesystem URL. An empty scheme would mean
+                // a URI that is just a path without any "proto:" part, which is still assumed a local path.
+                // A "file:" URI with a non-empty host would signify a remote URL, in which case we don't check if the
+                // file exists.
+                uri = normalizeLocalPath(Path.of(uri.getSchemeSpecificPart())).toUri();
```

Comment:
on Windows the file URI will looks something like `file:/D:foo/bar`, and `uri.getSchemeSpecificPart()` will include the leading slash resulting in `/D:foo/bar` which is an invalid path

Reply:
I think I resolved this by using `Path.of(uri)` to transform the URI into a Path, which should handle this gracefully.